schema.json

```diff
@@ -0,0 +1,39 @@
+{
+    "name": "xyz",
+    "version": "0.0.1",
+    "resources": {
+        "xyz:index:Random": {
+            "properties": {
+                "length": {
+                    "type": "integer"
+                },
+                "result": {
+                    "type": "string"
+                }
+            },
+            "required": [
+                "length",
+                "result"
+            ],
+            "inputProperties": {
+                "length": {
+                    "type": "integer"
+                }
+            },
+            "requiredInputs": [
+                "length"
+            ],
+            "stateInputs": {
```

Comment:
The doc comments for the schema imply that `stateInputs` should be optional - but:
- Python code generation fails if this is missing
- NodeJS generated code fails at runtime if this doesn't have the same set of properties as `properties`

I assume we should remove both of those requirements?

Reply:
Yes, we should remove both of those requirements.

Reply:
I hit this with the native Azure provider (I don't need state there yet). Do we have an issue tracking this?

Reply:
This appears to have been resolved in the latest pulumi/pulumi.